### DISCUSSION TITLE
Error when running lidar segmentation

### DISCUSSION THREAD

**Discussion by nealresearch** [opengeos/whiteboxR#92] [General]
I am trying to segment the lidar data. But i keep getting errors "Error in as.list.default(X) : 
  no method for coercing this S4 class to a vector". Also what is the output suppose to be. Is it an path or? 

Here is the code that i tried with

library("lidR")
library(whitebox)

las <-readLAS("M:\\build.las")

output <- "M:\\lidar\\las\\output.las"

segment <- wbt_lidar_segmentation(
  las,
  output,
  radius = 2,
  num_iter = 50,
  num_samples = 10,
  threshold = 0.15,
  model_size = 15,
  max_slope = 80,
  norm_diff = 10,
  maxzdiff = 1,
  classes = FALSE,
  ground = FALSE,
  wd = NULL,
  verbose_mode = FALSE,
  compress_rasters = FALSE,
  command_only = FALSE
)

**Comment by brownag** [comment]
{whitebox} does not currently support using S4 classes as input. In this case you are trying to use a class that represents a LAS file from the {lidR} package, instead try passing the path to the .las file as your input, e.g.
```r 
las <- "M:\build.las"
```

EDIT: As with `input`, `output` should also be a path to a LiDAR file (.las)
